taichi/codegen/llvm/codegen_llvm.cpp

```diff
@@ -448,46 +554,93 @@ void TaskCodeGenLLVM::visit(UnaryOpStmt *stmt) {
 #undef UNARY_INTRINSIC
 }
 
+void TaskCodeGenLLVM::create_elementwise_binary(
+    BinaryOpStmt *stmt,
+    std::function<llvm::Value *(llvm::Value *lhs, llvm::Value *rhs)> f) {
+  TI_ASSERT(stmt->lhs->ret_type->is<TensorType>());
+  TI_ASSERT(stmt->rhs->ret_type->is<TensorType>());
+  auto lhs_ty = stmt->lhs->ret_type->cast<TensorType>();
+  auto rhs_ty = stmt->rhs->ret_type->cast<TensorType>();
+  TI_ASSERT(lhs_ty->get_num_elements() == rhs_ty->get_num_elements());
+  auto lhs_vec = llvm_val[stmt->lhs];
+  auto rhs_vec = llvm_val[stmt->rhs];
+  auto elt_type_name = data_type_name(lhs_ty->get_element_type());
+  llvm::Value *result =
+      llvm::UndefValue::get(tlctx->get_data_type(stmt->ret_type));
+  for (int i = 0; i < lhs_ty->get_num_elements(); ++i) {
+    auto lhs = builder->CreateExtractElement(lhs_vec, i);
+    auto rhs = builder->CreateExtractElement(rhs_vec, i);
+    result = builder->CreateInsertElement(result, f(lhs, rhs), i);
+  }
+  llvm_val[stmt] = result;
+}
+
 void TaskCodeGenLLVM::visit(BinaryOpStmt *stmt) {
   auto op = stmt->op_type;
   auto ret_type = stmt->ret_type;
 
   if (op == BinaryOpType::add) {
-    if (is_real(stmt->ret_type)) {
+    if (is_real_tensor(stmt->ret_type) || is_real(stmt->ret_type)) {
       llvm_val[stmt] =
           builder->CreateFAdd(llvm_val[stmt->lhs], llvm_val[stmt->rhs]);
     } else {
       llvm_val[stmt] =
           builder->CreateAdd(llvm_val[stmt->lhs], llvm_val[stmt->rhs]);
     }
   } else if (op == BinaryOpType::sub) {
-    if (is_real(stmt->ret_type)) {
+    if (is_real_tensor(stmt->ret_type) || is_real(stmt->ret_type)) {
       llvm_val[stmt] =
           builder->CreateFSub(llvm_val[stmt->lhs], llvm_val[stmt->rhs]);
     } else {
       llvm_val[stmt] =
           builder->CreateSub(llvm_val[stmt->lhs], llvm_val[stmt->rhs]);
     }
   } else if (op == BinaryOpType::mul) {
-    if (is_real(stmt->ret_type)) {
+    if (is_real_tensor(stmt->ret_type) || is_real(stmt->ret_type)) {
       llvm_val[stmt] =
           builder->CreateFMul(llvm_val[stmt->lhs], llvm_val[stmt->rhs]);
     } else {
       llvm_val[stmt] =
           builder->CreateMul(llvm_val[stmt->lhs], llvm_val[stmt->rhs]);
     }
   } else if (op == BinaryOpType::floordiv) {
-    if (is_integral(ret_type))
-      llvm_val[stmt] =
-          create_call(fmt::format("floordiv_{}", data_type_name(ret_type)),
-                      {llvm_val[stmt->lhs], llvm_val[stmt->rhs]});
-    else {
+    if (is_integral_tensor(ret_type) || is_integral(ret_type)) {
```

Comment:
`floordiv` codegen has been removed in #6135. You need to modify https://github.com/taichi-dev/taichi/blob/79f2a6a9d692d81a6a1b46d7391d58ecf5aed771/taichi/transforms/demote_operations.cpp#L41.